### PR TITLE
Fix findcommand bug

### PR DESCRIPTION
**Pull request by Noob-No-1**
Fix bugs of FIndCommand listing guests with partial match.

### CODE REVIEW

File: src/main/java/seedu/guestnote/model/guest/FieldContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -36,7 +36,8 @@ public boolean test(Guest guest) {
             return arrayContainsKeyword(arr, keywords);
         } else {
             String fieldString = fieldValue.toString().toLowerCase();
-            return keywords.stream().anyMatch(keyword -> fieldString.contains(keyword.toLowerCase()));
+            return keywords.stream()
+                    .anyMatch(keyword -> fieldString.equalsIgnoreCase(keyword.trim()));
```
Comment (salads-source) [inline src/main/java/seedu/guestnote/model/guest/FieldContainsKeywordsPredicate.java:40]: Good use of `equalsIgnoreCase` to ensure that exact matches are found more reliably

File: src/main/java/seedu/guestnote/model/guest/FieldContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -66,12 +69,9 @@ private boolean arrayContainsKeyword(Object[] arr, List<String> keywords) {
...
         String combined = sb.toString();
-        for (String keyword : keywords) {
-            if (combined.contains(keyword.toLowerCase())) {
-                return true;
-            }
-        }
-        return false;
+        // Join all keywords with a space to form the full search phrase.
+        String joinedKeywords = String.join(" ", keywords).toLowerCase();
+        return combined.contains(joinedKeywords);
```
Comment (salads-source) [inline src/main/java/seedu/guestnote/model/guest/FieldContainsKeywordsPredicate.java:74]: Good that full phrases are matched accurately

File: src/test/java/seedu/guestnote/model/guest/FieldContainsKeywordsPredicateTest.java [hunk 3]
```diff
@@ -61,16 +61,11 @@ public void test_fieldContainsKeywords_returnsTrue() {
...
-        // Multiple keywords for a simple field (name)
-        predicate = new FieldContainsKeywordsPredicate<>(
-                guest -> guest.getName().toString(), Arrays.asList("Alice", "Bob"));
-        assertTrue(predicate.test(createGuest("Alice Bob")));
+        assertTrue(predicate.test(createGuest("Alice")));
 
         // Mixed-case keywords for a simple field
         predicate = new FieldContainsKeywordsPredicate<>(
-                guest -> guest.getName().toString(), Arrays.asList("aLIce", "bOB"));
+                guest -> guest.getName().toString(), Arrays.asList("aLIce bOB"));
```
Comment (salads-source) [inline src/test/java/seedu/guestnote/model/guest/FieldContainsKeywordsPredicateTest.java:68]: Would it be good to consider adding unit tests where keywords have leading or trailing spaces?

File: src/test/java/seedu/guestnote/model/guest/AnyFieldContainsKeywordsPredicateTest.java [hunk 4]
```diff
@@ -42,8 +42,8 @@ public void test_anyFieldContainsKeywords_returnsTrueWhenNameMatches() {
         AnyFieldContainsKeywordsPredicate compositePredicate =
                 new AnyFieldContainsKeywordsPredicate(Arrays.asList(namePredicate, emailPredicate));
 
-        // Create a guest whose name contains "Alice" (email is dummy).
-        Guest guest = createGuest("Alice Wonderland", "dummy@example.com", "Need wifi");
+        // Create a guest whose name contains "Alice", and the email is the predicate.
+        Guest guest = createGuest("Alice Wonderland", "alice@example.com", "Need wifi");
```
Comment (salads-source) [inline src/test/java/seedu/guestnote/model/guest/AnyFieldContainsKeywordsPredicateTest.java:46]: Good use of a more relevant email as compared to a dummy email address

**Review by salads-source — APPROVED** [review]
Nice fix for the partial matching logic. Consider adding a few more test cases for testing of edge cases
